### PR TITLE
Task completed: GP-111. Damage system

### PR DESCRIPTION
**Pull request by Kjusok**
[WIP] Damage system - https://lighteatersstudio.atlassian.net/browse/GP-111

### CODE REVIEW

File: Assets/Scripts/Gameplay/Character/CharacterStats.cs [hunk 1]
```diff
@@ -6,15 +6,14 @@ namespace Gameplay
     public class CharacterStats
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Character/CharacterStats.cs]: CharacterStats must implemented IStat interface

**Review by ega-ega — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Gameplay/Character/CharacterStats.cs (hunk 1 above)
Comment (Kjusok) [reply Assets/Scripts/Gameplay/Character/CharacterStats.cs]: Done

File: Assets/Scripts/Gameplay/Damage/AttackInfo.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿namespace Gameplay
+{
+    public class AttackInfo : IAttackInfo
+    {
+        public float Damage { get; }
+        public TypeDamage TypeDamage { get; }
+
+        
+        public AttackInfo(float damage, TypeDamage fire)
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Damage/AttackInfo.cs]: pls rename parametr to typeDamage

File: Assets/Scripts/Gameplay/Damage/DamageCalculator.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        /// Temporary solution
+        /// </summary>
+        public float Calculate(IAttackInfo attackInfo, IStats stats)
+        {
+            if (!stats.Imunne)
+            {
+                GetDamage(attackInfo);
+            }
+
+            return GetDamage(attackInfo);
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Damage/DamageCalculator.cs:12]: Its mistake, this return  must return zero value

File: Assets/Scripts/Gameplay/Damage/DamageCalculator.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
+﻿namespace Gameplay
+{
+    public class DamageCalculator : IDamageCalculator
+    {
+        /// <summary>
+        /// Temporary solution
```
Comment (ega-ega) [inline Assets/Scripts/Gameplay/Damage/DamageCalculator.cs]: pls remove anotation

**Review by ega-ega — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Gameplay/Damage/AttackInfo.cs (hunk 2 above)
Comment (Kjusok) [reply Assets/Scripts/Gameplay/Damage/AttackInfo.cs]: Done

File: Assets/Scripts/Gameplay/Damage/DamageCalculator.cs (hunk 3 above)
Comment (Kjusok) [reply Assets/Scripts/Gameplay/Damage/DamageCalculator.cs:12]: Done

File: Assets/Scripts/Gameplay/Damage/DamageCalculator.cs (hunk 4 above)
Comment (Kjusok) [reply Assets/Scripts/Gameplay/Damage/DamageCalculator.cs]: Done

**Review by ega-ega — APPROVED** [review]

**Pull request merged**